apps/designer/app/canvas/features/text-editor/interop.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  type ElementNode,
+  $getRoot,
+  $createTextNode,
+  $createParagraphNode,
+  TextNode,
+  LineBreakNode,
+  ParagraphNode,
+} from "lexical";
+import { $createLinkNode, LinkNode } from "@lexical/link";
+import type { ChildrenUpdates, Instance } from "@webstudio-is/react-sdk";
+import { createInstanceId } from "~/shared/tree-utils";
+
+// Map<nodeKey, Instance>
+export type Refs = Map<string, Instance>;
+
+const $writeUpdates = (
+  node: ElementNode,
+  updates: ChildrenUpdates,
+  refs: Refs
+) => {
+  const children = node.getChildren();
+  for (const child of children) {
+    if (child instanceof ParagraphNode) {
+      if (updates.length !== 0) {
+        updates.push("\n");
+      }
+      $writeUpdates(child, updates, refs);
+    }
+    if (child instanceof LineBreakNode) {
+      if (updates.length !== 0) {
+        // @todo should we visually distinct line breaks and paragraphs?
+        updates.push("\n");
+      }
+    }
+    if (child instanceof LinkNode) {
+      // @todo support link children styling
```

Comment:
Basically we can't make children of a link text bold right now, right?

Reply:
Not yet, same in current implementation. Need to change updaters structure. Will do in separate PR. Now just basic editing support.